pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -1,67 +1,68 @@
 { stdenv, fetchurl, openssl, python2, zlib, libuv, v8, utillinux, http-parser
-, pkgconfig, runCommand, which, libtool
-, version
-, sha256 ? null
-, src ? fetchurl { url = "https://nodejs.org/download/release/v${version}/node-v${version}.tar.xz"; inherit sha256; }
-, preBuild ? ""
+, pkgconfig, runCommand, which, libtool, fetchpatch
+, callPackage
+, darwin ? null
 , extraConfigFlags ? []
-, extraBuildInputs ? []
-, patches ? [],
- ...
+, lib
 }:
 
-assert stdenv.system != "armv5tel-linux";
-
 let
 
-  deps = {
-    inherit openssl zlib libuv;
-  } // (stdenv.lib.optionalAttrs (!stdenv.isDarwin) {
-    inherit http-parser;
-  });
+  inherit (darwin.apple_sdk.frameworks) CoreServices ApplicationServices;
+  inherit (stdenv.lib) optionals optionalAttrs concatMap maintainers licenses platforms;
+
+  sharedLibDeps = { inherit openssl zlib libuv; } // (stdenv.lib.optionalAttrs (!stdenv.isDarwin) { inherit http-parser; });
 
-  sharedConfigureFlags = name: [
+  sharedConfigureFlags = concatMap (name: [
     "--shared-${name}"
-    "--shared-${name}-includes=${builtins.getAttr name deps}/include"
-    "--shared-${name}-libpath=${builtins.getAttr name deps}/lib"
-  ];
+    "--shared-${name}-includes=${builtins.getAttr name sharedLibDeps}/include"
+    "--shared-${name}-libpath=${builtins.getAttr name sharedLibDeps}/lib"
+  ]) (builtins.attrNames sharedLibDeps);
 
-  inherit (stdenv.lib) concatMap optional optionals maintainers licenses platforms;
+  isNpmDisabled = flags: builtins.elem "--without-npm" flags;
 
-in stdenv.mkDerivation {
+in
 
-  inherit version src preBuild;
+  rec {
 
-  name = "nodejs-${version}";
+    buildInputs = stdenv.lib.optionals stdenv.isDarwin [ CoreServices ApplicationServices ]
+    ++ [ python2 which zlib libuv openssl ]
```

Comment:
This is what I was wondering about. If we no longer have npm, do we need `python2` here or can it be optional as well?